docs/batch_request.md

```diff
@@ -0,0 +1,30 @@
+## BatchRequest
+
+BatchRequest is useful when you want to make multiple requests efficiently. It batches all requests (upto 20 requests) into a json object and makes one api call. You can learn more about it on [Microsoft Docs](https://docs.microsoft.com/en-us/graph/json-batching). 
+
+## Code Sample
+
+```go
+import "github.com/microsoftgraph/msgraph-sdk-go-core"
+import abstractions "github.com/microsoft/kiota-abstractions-go"
+
+reqInfo := client.Me().CreateGetRequestInformation()
+batch := msgraphsdkcore.NewBatchRequest()
+batchItem := batch.AppendBatchItem(*reqInfo)
+
+resp, err := msgraphsdkcore.SendBatch(*batch, reqAdapter)
+
+// print the first response
+fmt.Println(resp.Responses[0].Body)
```

Comment:
I think that having access to the raw response is a first step, but it's probably not enough.
The other SDKs allow you to get the deserialized model from a batch response and we should probably offer the same thing.